### ISSUE TITLE
Notes for while I'm gone

### ISSUE THREAD

**Issue by ericaddison** [ericaddison/Hailing_Frequencies#23]
When joining a conversation, you `POST` to `/api/conversations/.../users/`. You should provide two `POST` arguments: `comm_option` and `comm_detail`. `comm_option` should be one of `'web'`, `'email'` or `'sms'`, and `comm_detail` should be a properly formatted email, phone number, or ??? (maybe nothing?) for web. The code for this is in `api/ConvUsers.py`. 

if no comm option is specified in the POST request, the `prefComm` option is used, which is currently `email` by default (this is done in `models/Users.py :: create()`).

**Comment by ericaddison** [comment]
OK!!!

Up all night! I think the API is in good shape (though you may find issues that need to be addressed as you work through the front end...). Here is a rundown of what we have going:

- Web API works for everything we could think of (get user info, get conv info, get messages, post/edit/delete messages, join conv, leave conv, etc)...
- The email version of the project actually works OK. If I create a new conversation now, it will email me a welcome message (because the current default `commOption` is email), and I can respond to that to post messages to the conversation.
- There is a file called `source.framework.communicate.py` where the message broadcast happens. That is, every time a message is created using the `source.api.ConvMessages.create_message()`, a message is broadcast to all participants. This is how messages in the real database should be created!
- To repeat that backwards ... a `ConvMessage` should be created by calling  `source.api.ConvMessages.create_message()`, which is how the web API and the email handler create them. That method calls the `broadcast_message()` method, which just takes a `ConvMessage` as an argument and sends the update out to all users in the conversation.

I'm not going to be able to put in too much more work when I get back ... so most of the rest is on you guys! @psigourney you have a great start on the web client! @mspagon I'm sure you'll do well with the Android!

Here are a couple more notes I have...

- To incorporate  Firebase interactive database into the broadcast process, you should fill in the `send_to_web()` method in `communicate.py` to write to the Firebase database. Currently that happens for every user in a loop, but I have a feeling that it might just need to happen once (so it will actually come out of the loop)...
- There is a new `config.authentication.py` that I NEED to send you... I set myself a reminder to do that when I get home from work. It has the SendGrid API key.

Here are some things I think we should be willing to drop, and leave as "future" features:
- SMS texting API (unless one of you are feeling ambitious)
- any of the more exotic conversation options (`revealOwner`, `viewAfterExpired`, etc)

Basically, if this turns into a hybrid email distribution list / chat room, I'll be happy :) 
I won't really have good internet for the next week, but please feel free to call me if you need help understanding my ridiculous code :) (phone number was in one of our early Canvas team conversations)

Thanks guys!!!

**Comment by psigourney** [comment]
Think I just borked my computer.  May need to reinstall OS.

**Comment by psigourney** [comment]
Could you send the authentication.py file?
